query/transformation.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2015 Square Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package query contains all the logic to parse
+// and execute queries against the underlying metric system.
+package query
+
+import (
+	"math"
+
+	"github.com/square/metrics/api"
+)
+
+// A transformation takes the list of values, and a "scale" factor:
+// It is the unitless quotient (sample duration / 1s)
+// So if the resolution is "once every 5 minutes" them we get a scale factor of 5*60 = 300
+type transformation func([]float64, transformationParameter) []float64
+
+type transformationParameter struct {
+	scale     float64
+	parameter float64
```

Comment:
Yes, changing to `[]value` for parameters makes sense with #43 merged.

I'm not sure what a transformation would do with the `Timerange`. Currently they use the `scale` which, as far as I can tell, is all they should care about (and `scale` is the seconds-duration of a single sample, so perhaps poorly named).